libs/src/sdk/api/tracks/TracksApi.ts

```diff
@@ -66,36 +64,43 @@ export class TracksApi extends TracksApiWithoutStream {
    * Upload a track
    */
   async uploadTrack(requestParameters: UploadTrackRequest) {
-    const { trackFile, coverArtFile, metadata, onProgress } = requestParameters
-
-    // Validate inputs
-    if (!isFileValid(trackFile)) {
-      throw new Error('Track file is not valid')
+    // Parse inputs
+    const result = createUploadTrackSchema().safeParse(requestParameters)
+    if (!result.success) {
+      throw new Error(`uploadTrack arguments not valid: ${result.error}`)
```

Comment:
Now that we have zod, it'd be good to have a util for standardizing our error handling/messages e.g.
`validateRequestParams(name, schema)` -> does lines 68-70 here

or can even just make a new type of Error that takes the method name and `result.error`